tcp-server/src/main.rs

```diff
@@ -0,0 +1,116 @@
+use std::borrow::BorrowMut;
+use std::collections::vec_deque::Iter;
+use std::collections::VecDeque;
+use std::io;
+use std::io::{BufRead, BufReader, Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+use std::thread;
+use std::thread::JoinHandle;
+use std::time::Duration;
+
+use redisish::{Command, parse};
+
+trait Mailbox {
+    /// TODO Is it fine that mutability is hidden? How can we do it some other way?
+    fn append(&self, email: String);
+    /// TODO does it make sense to return an abstract iterator and how can this be achieved?
+    fn list_emails(&self) -> String;
```

Comment:
out of scope for this exercise; the [leveldb trail](https://exercises-2021.ferrous-systems.com/ffi-1-exercise.html) has a [section](https://exercises-2021.ferrous-systems.com/ffi-3-exercise.html) on iterators